module/src/components/modal/modal.component.tsx

```diff
@@ -70,56 +75,64 @@ export const Modal = React.forwardRef<HTMLDivElement, IModalProps>(
       wrapperClassName,
       disableClose,
       closeTime,
+      onMouseDown,
+      onClose,
       ...nativeProps
     },
     ref
   ) => {
+    const wrapperRef = useModalLayerElement();
+
     const internalRef = React.useRef<HTMLDivElement>(null);
     React.useImperativeHandle(ref, () => internalRef.current!);
 
-    const close = React.useCallback(() => {
-      if (!disableClose) {
-        onOpenChange?.(false);
-      }
-    }, [onOpenChange, disableClose]);
+    const close = React.useCallback(() => ModalUtils.closeModal({ disableClose, onClose, onOpenChange }), [onOpenChange, disableClose, onClose]);
 
-    /** Close when the user clicks outside of the dropdown */
-    const onWindowClick = React.useCallback(
-      (event: React.MouseEvent<HTMLElement>) => {
-        if (isOpen && closeOnWindowClick && (!internalRef.current || !DOM.clickIsInsideElement(internalRef.current, event))) {
-          close();
-        }
+    /** ensure that clicks initiated inside the modal don't count as clicks on the wrapper if they end outside it (i.e. if a user drags the mouse from inside the modal to the background, it'll still count as a click on that background) */
+    const [mouseDownIsInsideModal, setMouseDownIsInsideModal] = React.useState(false);
+    const onMouseDownModal = React.useCallback(
+      (event: React.MouseEvent<HTMLDivElement>) => {
+        onMouseDown?.(event);
+        setMouseDownIsInsideModal(true);
       },
-      [isOpen, close, closeOnWindowClick]
+      [onMouseDown]
     );
 
+    /** Close when the user clicks outside of the dropdown */
+    const onWindowClick = React.useCallback(() => {
```

Comment:
It might make more sense to name this `onDocumentClick`, window and document aren't the same thing.